site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,190 @@
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Row-Level Deletes
+
+Iceberg supports metadata-based deletion through the `DeleteFiles` interface.
+It allows you to quickly delete a specific file or any file that might match a given expression without the need to read or write any data in the table.
+
+Row-level deletes target more complicated use cases such as general data protection regulation (GDPR).
+Copy-on-write and merge-on-read are two different approaches to handle row-level delete operations. Here are their definitions in Iceberg:
+
+- **copy-on-write**: a delete directly rewrites all the affected data files.
+- **merge-on-read**: delete information is encoded in the form of _delete files_. The table reader can apply all delete information at read time.
+
+Overall, copy-on-write is more efficient in reading data, whereas merge-on-read is more efficient in writing deletes, but requires more maintenance and tuning to be performant in reading data with deletes.
+Users can choose to use **both** copy-on-write and merge-on-read for the same Iceberg table based on different situations. 
+For example, a time-partitioned table can have newer partitions maintained with the merge-on-read approach through a streaming pipeline,
+and older partitions maintained with the copy-on-write approach to apply less frequent GDPR deletes from batch ETL jobs.
+
+There are use cases that could only be supported by one approach such as change data capture (CDC).
+There are also limitations for different compute engines that lead them to prefer one approach over another.
+Please check out the documentation of the specific compute engines to see the details of their capabilities related to row-level deletes.
+This article will focus on explaining Iceberg's core design of copy-on-write and merge-on-read.
+
+!!!Note
+    Update is modeled as a delete with an insert within the same transaction in Iceberg, so this article only explains delete-related concepts. 
+
+## Copy-on-write
+
+In the copy-on-write approach, given a user's delete requirement, the write process would search for all the affected data files and perform a rewrite operation.
+
+For example, consider an unpartitioned table with schema `(id bigint, category string, data string)` that has the following files:
+
+```
+file A: (1, 'c1', 'data1'), (2, 'c1', 'data2')
+file B: (3, 'c2', 'data1'), (4, 'c2', 'data2')
+file C: (5, 'c3', 'data3'), (6, 'c3', 'data2')
```

Comment:
Fake data like `c1`, `c2`, etc. is really, really hard to follow. I suggest using real words or names so that it is clear what's happening. The spec has an example of different bears that you can use here. That is easy to follow:

 1: id | 2: category | 3: name
-------|-------------|---------
 1     | marsupial   | Koala
 2     | toy         | Teddy
 3     | ursidae     | Grizzly
 4     | ursidae     | Polar
 5     | candy        | Gummi